docs/modules/ROOT/pages/how-tos/configuring/configuration-as-code.adoc

```diff
@@ -1,4 +1,322 @@
+
+NOTE: Throughout this document, substitute your application and component names for placeholder values like `application-a`, `application-a`, `component-a`, etc.
+
+== Create a File Structure
+As with many Kustomize projects, we'll need a space to define a `base` and some `overlay`(s).  Within our `overlay`(s), we'll also define a folder for a base _set_ of components and a folder for each application-specific overrides for that base set of components.
+
+Create a file structure as follows:
+[source,bash]
+----
+├── base
+│   ├── component.yaml
+│   └── kustomization.yaml
+├── overlay
+│   └── application-a
+│       ├── base
+│       │   ├── application.yaml
+│       │   ├── component-a
+│       │   │   ├── component-a-override.yaml
+│       │   │   └── kustomization.yaml
+│       │   ├── component-b
+│       │   │   ├── component-b-override.yaml
+│       │   │   └── kustomization.yaml
+│       │   └── kustomization.yaml
+│       └── v1-overlay
+│           ├── application-patch.yaml
+│           ├── component-patch.yaml
+│           ├── exception-component-patch-1.yaml
+│           └── kustomization.yaml
+└── README.md
+----
+
+NOTE: You can omit `component-b` if you only wish to define a single component.  This was included for illustration. 
+
+In this file structure, you will find:
+. `base`: which is a directory housing your base component yaml and a kustomize to point to the base.
+. `overlay`: which contains one or more application variants, useful when defining multiple different applications.
+. `overlay/application-a/base`: a base which defines one or more component overrides, one per component in the application and a base for your application.
+. `overlay/application-a/v1-overlay`: An override for your application and an application specific patches for your component(s).
+
+== Create a Base for your Components
+If you are creating more than one component, its likely that your components will share some, if not most, of their configuration. These default or global values can be defined in the Base component and overridden on a case-by-case basis in your Component overlays.  
+
+=== Creating your Base Component
+
+You'll define this base in `base/component.yaml`.  It typically follows a pattern similar to the following:
+[source,yaml]
+----
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: Component
+metadata:
+  annotations:
+    image.redhat.com/generate: "true"
+    appstudio.openshift.io/pac-provision: request
+    build.appstudio.openshift.io/request: configure-pac
+  name: example-component
+spec:
+  componentName: example-component
+  application: appName
+  targetPort: 8080
+  source:
+      git:
+        url: gitUrl
+        context: ./
+        dockerfileUrl: ContainerFileLocation
+        revision: defaultBranch
+----
+
+NOTE: This will create a component with a custom build pipeline.  If you wish to use the default build, omit the `build.appstudio.openshift.io/request: configure-pac` annotation.
```

Comment:
I don't think we need the second sentence. Is the first consistent with other pages?